samples/test/util.py

```diff
@@ -1,55 +1,104 @@
 import kfp
+import os
+from typing import Dict, List, Callable
+from dataclasses import dataclass
 
 MINUTE = 60
 
 
-def default_verify_func(run_id, run):
+def _default_verify_func(run_id, run):
     assert run.status == 'Succeeded'
 
 
-def run_pipeline_func(
-    pipeline_func,
-    verify_func=default_verify_func,
-    mode=kfp.dsl.PipelineExecutionMode.V2_COMPATIBLE
-):
+@dataclass
+class TestCase:
+    '''Test case for running a KFP sample'''
+    pipeline_func: Callable
+    mode: kfp.dsl.PipelineExecutionMode = kfp.dsl.PipelineExecutionMode.V2_COMPATIBLE
+    arguments: Dict[str, str] = None
+    verify_func: Callable = _default_verify_func
+
+
+def run_pipeline_func(test_cases: List[TestCase]):
     """Run a pipeline function and wait for its result.
 
     :param pipeline_func: pipeline function to run
     :type pipeline_func: function
     """
 
+    def test_wrapper(run_pipeline):
+        for case in test_cases:
+            result = run_pipeline(
+                pipeline_func=case.pipeline_func,
+                mode=case.mode,
+                arguments=case.arguments or {}
+            )
+            case.verify_func(run=result, run_id=result.id)
+
+    _run_test(test_wrapper)
+
+
+def _run_test(callback):
+
     def main(
-        output_directory: str,  # example gs://your-bucket/path/to/workdir
-        host: str = 'http://ml-pipeline:8888',
+        output_directory: str = None,  # example
```

Comment:
nit: type these as Optional[str] since None is a valid value.

Reply:
Thank you! Now I know python typings better